source/agora/node/FullNode.d

```diff
@@ -416,8 +416,16 @@ public class FullNode : API
 
     protected bool acceptBlock (const ref Block block) @trusted
     {
+        ExpiringValidator[] ex_validators;
+        this.enroll_man.getExpiringValidators(block.header.height, ex_validators);
         // Attempt to add block to the ledger (it may be there by other means)
-        this.ledger.acceptBlock(block);
+        if (this.ledger.acceptBlock(block))
+        {
+            ex_validators[].each!(ex => this.network.unwhitelist(ex.pubkey));
```

Comment:
You don't actually need to slice an array, it's already a slice. :)

Just curious, do you do it to make code more readable?

Reply:
It's just me being a bad D programmer:)